code/datums/components/links_to_operating_computers.dm

```diff
@@ -0,0 +1,126 @@
+		stack_trace("Parent must be a movable")
+		return COMPONENT_INCOMPATIBLE
+
+	if (!isnull(provide_upgraded_surgeries))
+		src.provide_upgraded_surgeries = provide_upgraded_surgeries
+
+/datum/component/links_to_operating_computers/RegisterWithParent()
+	RegisterSignal(parent, COMSIG_MOVABLE_MOVED, PROC_REF(on_movable_moved))
+	register_to_loc()
+
+/datum/component/links_to_operating_computers/UnregisterFromParent()
+	var/atom/atom_parent = parent
+
+	UnregisterSignal(atom_parent, COMSIG_MOVABLE_MOVED)
+	unregister_loc(atom_parent.loc)
+
+	for (var/patient in patients)
+		SEND_SIGNAL(src, COMSIG_LINKS_TO_OPERATING_COMPUTERS_PATIENT_REMOVED, patient)
+
+	patients.Cut()
+
+/datum/component/links_to_operating_computers/proc/on_atom_entered(source, mob/living/carbon/enterer)
+	SIGNAL_HANDLER
+
+	if (enterer in patients)
+		return
+
+	check_atom_on_object(enterer)
+
+/datum/component/links_to_operating_computers/proc/check_atom_on_object(atom)
+	if (!iscarbon(atom))
+		return
+
+	RegisterSignal(atom, COMSIG_LIVING_SET_BODY_POSITION, PROC_REF(on_enterer_body_position_changed))
+	check_patient(atom)
+
+/datum/component/links_to_operating_computers/proc/on_atom_exited(source, mob/living/carbon/exiter)
+	SIGNAL_HANDLER
+
+	if (!istype(exiter))
+		return
+
+	UnregisterSignal(exiter, COMSIG_LIVING_SET_BODY_POSITION)
+	remove_patient(exiter)
+
+/datum/component/links_to_operating_computers/proc/on_enterer_body_position_changed(mob/living/carbon/source)
+	SIGNAL_HANDLER
+	check_patient(source)
+
+/datum/component/links_to_operating_computers/proc/check_patient(mob/living/carbon/patient)
+	if (patient.body_position == LYING_DOWN && !(patient in patients))
+		patients += patient
+		SEND_SIGNAL(src, COMSIG_LINKS_TO_OPERATING_COMPUTERS_PATIENT_ADDED, patient)
+	else if (patient.body_position == STANDING_UP)
+		remove_patient(patient)
+
+/datum/component/links_to_operating_computers/proc/remove_patient(mob/living/carbon/patient)
+	if (!(patient in patients))
+		return
+
+	patients -= patient
+	SEND_SIGNAL(src, COMSIG_LINKS_TO_OPERATING_COMPUTERS_PATIENT_REMOVED, patient)
+
+/datum/component/links_to_operating_computers/proc/on_operating_computer_initialized(datum/source, obj/machinery/computer/operating/operating_computer)
+	SIGNAL_HANDLER
+	initialize_to_operating_computer(operating_computer)
+
+/datum/component/links_to_operating_computers/proc/initialize_to_operating_computer(obj/machinery/computer/operating/operating_computer)
+	// Catch the new operating computer up to speed
+	SEND_SIGNAL(operating_computer, COMSIG_LINKS_TO_OPERATING_COMPUTERS_INITIALIZED, src, patients)
+
+/datum/component/links_to_operating_computers/proc/on_movable_moved(datum/source, atom/old_loc)
+	SIGNAL_HANDLER
+
+	unregister_loc(old_loc)
+	register_to_loc()
+
+/datum/component/links_to_operating_computers/proc/register_to_loc()
+	var/atom/atom_parent = parent
+	var/atom/loc = atom_parent.loc
+
+	for (var/atom/atom as anything in loc)
+		check_atom_on_object(atom)
+
```

Comment:
this name kinda sucks

Reply:
Any ideas? It's checking if the atom on the current loc is eligible for being a patietn

Reply:
potential_patient maybe? s a bit verbose but idk.
check_atom_on_object is also not great, it's attempting to make something on our tile a patient. try_add_patient maybe?

Reply:
Oh you mean the variable sure